### PR TITLE
Student: MOTD: include only body of fetched HTML resource file #7794

### LINKED ISSUE #7794: Student: MOTD: include only body of fetched HTML resource file
**Issue:**
Unrelated content like `meta tag`, `favicon tag` and `title tag` found inside `student-motd` in the Student submit feedback page. _[screenshot]_

screenshot from 2017-07-19 02-52-26

**Reason:**
Looked into the code and found that `student-motd` get's its data from [`fetchMotd`](https://github.com/TEAMMATES/teammates/blob/master/src/main/webapp/dev/js/main/studentMotd.es6#L4) function via `GET` request.

On printing, the data that's `student-motd` get's from [`fetchMotd`](https://github.com/TEAMMATES/teammates/blob/master/src/main/webapp/dev/js/main/studentMotd.es6#L4) function is _[screenshot]_
screenshot from 2017-07-19 03-39-34

**Solution:**
We should modify the [`fetchMotd`](https://github.com/TEAMMATES/teammates/blob/master/src/main/webapp/dev/js/main/studentMotd.es6#L4) function to this 
```javascript
function fetchMotd(motdUrl, motdContentSelector, motdContainerSelector) {
    $.ajax({
        type: 'GET',
        url: `${window.location.origin}/${motdUrl}`,
        success(data) {
            const Bodydata = data.replace('<body', '<body><div id="body"')
                                 .replace('</body>','</div></body>');
            $(motdContentSelector).html($(Bodydata).filter('#body'));
            $(motdContentSelector).html(data);
        },
        error() {
            $(motdContainerSelector).html('');
        },
    });
}
```
This updated `fetchMotd` on success will only return the data present inside the `Body element` to `student-motd` container.

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #7794 

Modified the [`fetchMotd`](https://github.com/TEAMMATES/teammates/blob/master/src/main/webapp/dev/js/main/studentMotd.es6#L4) function to this 
```javascript
function fetchMotd(motdUrl, motdContentSelector, motdContainerSelector) {
    $.ajax({
        type: 'GET',
        url: `${window.location.origin}/${motdUrl}`,
        success(data) {
            const bodyContent = data.match(/<body[^>]*>[\s\S]*<\/body>/gi);
            $(motdContentSelector).html(bodyContent);
        },
        error() {
            $(motdContainerSelector).html('');
        },
    });
}
```
This updated `fetchMotd` on success will only return the data present inside the `Body element` to `student-motd` container.

### CODE REVIEW

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/dev/js/main/studentMotd.es6 [hunk 1]
```diff
@@ -6,7 +6,8 @@ function fetchMotd(motdUrl, motdContentSelector, motdContainerSelector) {
         type: 'GET',
         url: `${window.location.origin}/${motdUrl}`,
         success(data) {
-            $(motdContentSelector).html(data);
+            const bodyContent = data.match(/<body[^>]*>[\s\S]*<\/body>/gi);
```
Comment (amarlearning) [inline src/main/webapp/dev/js/main/studentMotd.es6]: @thenaesh I know `.` is used to match all character but due to some reason, it doesn't seem to work. I checked the regular expression on [Regexr](http://regexr.com/) the `.` notation regular expression isn't working while the `\s\S` works fine.

_[Regex matching screenshot]_: `<body[^>]*>[\s\S]*<\/body>`
screenshot from 2017-07-21 14-28-59

_[Not matching Regex]_: `<body[^>]*>.*<\/body>`
screenshot from 2017-07-21 14-33-04

Comment (thenaesh) [inline src/main/webapp/dev/js/main/studentMotd.es6]: It turns out that `.` matches everything **except line breaks**. That's why it doesn't work. So I guess we can just go with `[\s\S]`.

**Comment by amarlearning** [comment]
@damithc reminder!

**Pull request merged**